buildenv/buildenv.nix

```diff
@@ -0,0 +1,278 @@
+          ''
+            cp ${builtins.toFile "hook-on-activate" hook."on-activate"} $out/activate.d/hook-on-activate
+          ''
+        else
+          ""
+      )
+      # service-config.yaml section
+      (
+        if (serviceConfigYaml != "") then
+          ''
+            cp ${/. + serviceConfigYaml} $out/activate.d/service-config.yaml
+          ''
+        else
+          ""
+      )
+    ]
+    ++ (
+      # [profile] section
+      builtins.map
+        (
+          i:
+          if (builtins.hasAttr i profile) then
+            let
+              f = builtins.toFile "profile-${i}" (builtins.getAttr i profile);
+            in
+            "cp ${f} $out/activate.d/profile-${i}\n"
+          else
+            ""
+        )
+        [
+          "bash"
+          "fish"
+          "tcsh"
+          "zsh"
+        ]
+    )
+    ++ (
+      # [build] section
+      builtins.map (
+        i:
+        let
+          b = builtins.getAttr i build;
+        in
+        (
+          if (builtins.hasAttr "command" b) then
+            (
+              let
+                f = builtins.toFile "build-${i}" (builtins.getAttr "command" b);
+              in
+              ''
+                mkdir -p $out/package-builds.d
+                cp ${f} $out/package-builds.d/${i}
+              ''
+            )
+          else
+            ""
+        )
+      ) (builtins.attrNames build)
+    );
+
+  createManifestScript = builtins.toFile "create-manifest-script" (
+    builtins.concatStringsSep "" createManifestChunks
+  );
+
+  # Create manifest package as derivation which invokes above script.
+  manifestPackage = builtins.derivation {
+    name = "manifest";
+    inherit system;
+    builder = "/bin/sh";
+    args = [
+      "-eux"
+      createManifestScript
+    ];
+  };
+
+  # Calculate inputSrcs by noting all storePaths for this system's
+  # packages found in the packages list.
+  inputSrcs = builtins.concatMap (
+    package:
+    if package.system == system then
+      (
+        if (builtins.hasAttr "outputs" package) then
+          (
+            # Important: report storePaths rather than strings because
+            # the updated string context populates `inputSrcs` for the
+            # resulting derivation.
+            let
+              outputsToInstall =
+                if (builtins.hasAttr "outputs_to_install" package) then
+                  (builtins.getAttr "outputs_to_install" package)
+                else if (builtins.hasAttr "outputs-to-install" package) then
+                  (
+                    # XXX kebab-case flake lock bug
+                    builtins.getAttr "outputs-to-install" package
+                  )
+                else
+                  null;
+              filteredOutputs =
+                if (outputsToInstall == null) then
+                  (builtins.attrValues package.outputs)
+                else
+                  (builtins.map (x: builtins.getAttr x package.outputs) outputsToInstall);
```

Comment:
blocking: I believe we're currently not respecting outputsToInstall
https://github.com/flox/flox/blob/e279be165461620ee12377bca0763f9e0c535f2f/pkgdb/src/realisepkgs/realise.cc#L159

If that's the case, we shouldn't change that in this PR

Reply:
Yes that's a behaviour change that would cause installing openssl as a library to fail since those are installed to `^dev` only which is not an `outputToInstall`. Unless we are able to and educate users i dont think we can release this yet.
Blocking for me are currently:

1. missing documentation _that outputs exist_,
2. how to know which outputs are there for any package 
3. how to specify outputs when installing (cli & manifest)

Reply:
Agreed, I think we should revisit all of those separately as part of #1363

Reply:
As discussed we shouldn't break the `outputsToInstall` support of this buildenv implementation, but rather update the rust CLI to ask for all outputs to be installed by default. That should be implemented in the followup issue #2207 rather than changing the implementation here.

And in future we can plan the feature change by which we guide people through the process of modifying their manifests to be more precise about exactly which outputs they need when installing a given package.

Reply:
Rust isn't in a position to influence the outcome after all. Updated the code to nobble the behaviour until we can plan a transition that fixes the problem in commit 9074dcb9eea84b6223cc7d80386ff256a902a32d. Resolving conversation.